src/api/telemetry/TelemetryCollection.js

```diff
@@ -296,7 +330,17 @@ export class TelemetryCollection extends EventEmitter {
                     datum => this.parseTime(datum)
                 );
                 added = this.futureBuffer.splice(0, endIndex);
-                this.boundedTelemetry = [...this.boundedTelemetry, ...added];
+
+                let combined = [...this.boundedTelemetry, ...added];
```

Comment:
This step is not necessary if strategy is latest. We don't want to combine boundedTelemetry and `added`. We just want the last member of `added` (ie the latest value from the future buffer).